src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -351,6 +351,7 @@ public static IResourceBuilder<TDestination> WithReference<TDestination>(this IR
         connectionName ??= resource.Name;
 
         builder.WithRelationship(resource, KnownRelationshipTypes.Reference);
+        (builder as IResourceBuilder<IResourceWithWaitSupport>)?.WaitFor(source);
```

Comment:
I need to make this a bit smarter; basically we may need to wait for some references if the referenced endpoint may not be allocated before the resources are created.

Reply:
Reverted some of these changes; will need to be part of a larger potential rework for a future release.